partner_risk_insurance/templates/invoice_risk_insurance_template.xml

```diff
@@ -0,0 +1,170 @@
+                    <td colspan="5">
+                        <strong>
+                            <t t-out="move.partner_id.country_id.code" />
+                            -
+                            <t t-out="move.partner_id.country_id.name" />
+                        </strong>
+                    </td>
+                </tr>
+            </t>
+            <tr>
+                <td t-out="move.name" style="border: 1px solid black" />
+                <td t-out="move.partner_id.ref or ''" style="border: 1px solid black" />
+                <td t-out="move.partner_id.name" style="border: 1px solid black" />
+                <td
+                    t-out="move.amount_total"
+                    t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                    style="border: 1px solid black"
+                    align="right"
+                >
+                </td>
+                <td t-out="move.date" style="border: 1px solid black" align="right"/>
+                <t
+                    t-set="insurance_country_total"
+                    t-value="insurance_country_total + move.amount_total"
+                />
+            </tr>
+            <t
+                t-if="move_last or move.partner_id.country_id != moves[move_index + 1].partner_id.country_id"
+            >
+                <tr>
+                    <td colspan="5" align="right">
+                        <strong >
+                            Total by country:
+                            <t
+                                t-out="insurance_country_total"
+                                t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                            >
+                            </t>
+                        </strong>
+                    </td>
+                </tr>
+                <t t-set="insurance_country_total" t-value="0" />
+            </t>
+        </t>
+        <tr>
+            <td colspan="5" align="right">
+                <strong>
+                    Total
+                    <t t-if="is_insured">insured</t>
+                    <t t-else="">not insured</t>
+                    customers:
+                    <t
+                        t-out="insurance_partial_total"
+                        t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                        align="right"
+                    >
+                    </t>
+                </strong>
+            </td>
+        </tr>
+    </template>
+    <template id="report_invoice_risk_insurance_document">
+        <t t-call="web.external_layout">
+
+            <t
+                t-set="insurance_total_amount"
+                t-value="sum(docs.mapped('amount_total'))"
+            />
+            <t t-set="change_insurance_type" t-value="False" />
+            <t
+                t-set="insured_moves"
+                t-value="docs.filtered(lambda mv: mv.partner_id.risk_insurance_grant_date)"
```

Comment:
Can't the same strategy be used as for countries instead of going through all the records several times to filter them?

Reply:
This way makes the code more understandable, but it is true that this consultations to record are unnecessary. It is prefearable to follow the same strategy for the entire report?